### ISSUE TITLE
Automatically clearing callbacks 

### ISSUE THREAD

**Issue by mzgreen** [Polidea/Cockpit#108]
**Describe the bug**
Library allows to pass a callback that will be invoked when a button in Cockpit menu is clicked. It's easy to create a memory leak with the current implementation.

**Library version**
Version of Cockpit the issue occurs in: `3.0.1`.

**To reproduce**
Add a callback:
```
// we're inside Activity's onCreate
Cockpit.addFooCallback {
    // access Activity methods or fields 
}
```
Such callback captures activity instance and is kept by Cockpit. If we rotate the device then we have a memory leak.
To avoid that we can do:
```
val fooCallback = ActionRequestCallback {
    // access Activity methods or fields 
}

// in onCreate
Cockpit.addFooCallback(fooCallback)

// in onDestroy
Cockpit.removeFooCallback(fooCallback)
```
Maybe it would be possible to automatically remove the callbacks when necessary? Or at least add a possibility to enable/disable automatic callback removal? I'm not 100% sure that we want to always remove them, it's a topic for discussion. Just wanted to leave it here.

**Comment by mikolak** [comment]
My first thought was to remove all notifications/callbacks when CockpitDialog is destroyed, since it will be destroyed along with parent activity, but:
- callback may be capturing a fragment, which may be destroyed long before the parent activity,
- CockpitDialog might be destroyed separately from the parent activity, which would force user to set up all the notifications and callbacks again each time the dialog is shown.

Perhaps the best way would be to save some context key for each callback/notification. Then each time a context is invalidated (activity/fragment destroyed) user would simple call something like `Cockpit.invalidateContext(this::class.toString())` in corresponding `onStop()`? 
This could be also extended/swapped to implement LifecycleObserver, though I don't have much experience with it and don't know if it indeed is an option.

**Comment by mikolak** [comment]
Proposed implementation ATM:
- `Cockpit.register(...)Callback(LifecycleOwner, Callback)`
- deprecate current method
- new `Cockpit.registerForever(...)()` method to make it explicit that it's unsafe for context capturing methods.

**Comment by martawoldanska** [comment]
Fixed in version 3.1.0.
